### ISSUE TITLE
Declaring constructors as methods is strange

### ISSUE THREAD

**Issue by domenic** [privacycg/saa-non-cookie-storage#31]
In https://privacycg.github.io/saa-non-cookie-storage/#storageaccesshandle the members `BroadcastChannel` and `SharedWorker` are presumably supposed to be sort of namespaced versions of the global `BroadcastChannel` and `SharedWorker` constructors.

However, since they are declared as Web IDL operations, they end up as JavaScript methods, not constructors.

This has several observable differences, notably around how they don't require `new`, and fail various [IsConstructor](https://tc39.es/ecma262/#sec-isconstructor) tests. E.g. they cannot be subclassed.

Ideally we would define these as constructors. Doing so would probably require new Web IDL infrastructure, however...

**Comment by annevk** [comment]
@johannhof I think this issue needs addressing. I think the easiest path forward here is to define these as `createBroadcastChannel()` or some such. I wouldn't be opposed to a more complicated story that also allows for subclassing, but these weird class-like-looking methods we should not enshrine in the platform.

**Comment by johannhof** [comment]
@annevk yeah, I agree with this feedback. We can prioritize proposing an alternative API surface. I think we can make this mostly backwards compatible for existing Chrome deployments, too.

cc @cfredric
